src/Surfnet/ServiceProviderDashboard/Application/Metadata/JsonGenerator.php

```diff
@@ -410,9 +404,36 @@ private function generateLogoMetadata(Entity $entity)
             $height = 50;
         }
 
-        $metadata['logo:0:width'] = (string) $width;
-        $metadata['logo:0:height'] = (string) $height;
+        $metadata['logo:0:width'] = (string)$width;
+        $metadata['logo:0:height'] = (string)$height;
 
         return $metadata;
     }
+
+
+    /**
+     * @param Entity $entity
+     * @return array
+     */
+    private function generateAclData(Entity $entity)
+    {
+        if ($entity->isIdpAllowAll()) {
+            return [
+                'allowedEntities' => [],
+                'allowedall' => true,
+            ];
+        }
+
+        $providers = [];
+        foreach ($entity->getIdpWhitelist() as $entityId) {
+            $providers[] = [
```

Comment:
What if we have no Idps in the list? Is this possible? And if it is the case, what should we do?

Reply:
That would be possible and then no entities are whitelisted.